lib/graphql/schema/printer.rb

```diff
@@ -3,41 +3,98 @@ module GraphQL
   class Schema
     # Used to convert your {GraphQL::Schema} to a GraphQL schema string
     #
-    # @example print your schema to standard output
+    # @example print your schema to standard output (via helper)
     #   MySchema = GraphQL::Schema.define(query: QueryType)
     #   puts GraphQL::Schema::Printer.print_schema(MySchema)
     #
-    module Printer
-      extend self
+    # @example print your schema to standard output
+    #   MySchema = GraphQL::Schema.define(query: QueryType)
+    #   puts GraphQL::Schema::Printer.new(MySchema).print_schema
+    #
+    # @example print a single type to standard output
+    #   query_root = GraphQL::ObjectType.define do
+    #     name "Query"
+    #     description "The query root of this schema"
+    #
+    #     field :post do
+    #       type post_type
+    #       resolve ->(obj, args, ctx) { Post.find(args["id"]) }
+    #     end
+    #   end
+    #
+    #   post_type = GraphQL::ObjectType.define do
+    #     name "Post"
+    #     description "A blog post"
+    #
+    #     field :id, !types.ID
+    #     field :title, !types.String
+    #     field :body, !types.String
+    #   end
+    #
+    #   MySchema = GraphQL::Schema.define(query: query_root)
+    #
+    #   printer = GraphQL::Schema::Printer.new(MySchema)
+    #   puts printer.print_type(post_type)
```

Comment:
I can't think of a good reason for someone to use one over the other.

```ruby
# via printer instance
printer = Graph::Schema::Printer.new(schema)
schema.types.each {|t| printer.print_type(t) }

# via to_def with a printer
printer = Graph::Schema::Printer.new(schema)
schema.types.each {|t| t.to_definition(schema, printer: printer) }

# via to_def with just a schema
schema.types.each {|t| t.to_definition(schema) }
```

I wanted to avoid instantiating a printer for each type but that's possible anyway with the optional `printer` param.